apps/emqx_connector/src/emqx_connector_http.erl

```diff
@@ -552,9 +558,12 @@ formalize_request(Method, BasePath, {Path, Headers, _Body}) when
 ->
     formalize_request(Method, BasePath, {Path, Headers});
 formalize_request(_Method, BasePath, {Path, Headers, Body}) ->
-    {filename:join(BasePath, Path), Headers, Body};
+    {prepend_basepath(BasePath, Path), Headers, Body};
 formalize_request(_Method, BasePath, {Path, Headers}) ->
-    {filename:join(BasePath, Path), Headers}.
+    {prepend_basepath(BasePath, Path), Headers}.
+
+prepend_basepath(BasePath, Path) ->
+    [BasePath, $/ | string:trim(Path, leading, [$/])].
```

Comment:
BasePath is often `"/"` by default, we will get "//path" here.
This is fixed in another PR so we may live `filename:join` here to avoid conflicts.